src/data/batch_buffer.py

```
@@ -0,0 +1,84 @@
+    def fill(self, token_ids: List[int], actual_length: Optional[int] = None):
+        """
+        Fill buffer with token IDs.
+        
+        Args:
+            token_ids: List of token IDs to fill buffer with
+            actual_length: Actual length to use (if different from max)
+        """
+        seq_length = actual_length if actual_length is not None else self.max_seq_length
+        
+        if len(token_ids) > self.batch_size:
+            raise ValueError(f"token_ids length {len(token_ids)} exceeds batch_size {self.batch_size}")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)